tutorials/rag/evaluate-rag-flow.ipynb

```diff
@@ -842,6 +832,277 @@
     "\n",
     "![Eval Logs table](../../assets/images/evaluate_rag_flow_logs.png)"
    ]
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "# Using Retriever as a Tool\n",
+    "\n",
+    "In the above example, the flow of our app is static - given the question we always call the retriever, pass the context to the model and then respond to the user. It can be helpful to have a more dynamic setup where we instead allow the model to choose when use the retriever.\n",
+    "This allows the model to do smart query expansion, or ask for clarifying questions, before deciding to retrieve the context which can be significant boost in performance.\n",
+    "\n",
+    "This entails updating the Prompt with a Tool schema definition for the retriever. And then routing your flow to the retriever based on when or not the model makes a tool call. The Flow on Humanloop can capture the full trace of events so that you have full transparency into the models actions. And the `evaluations.run(...)` utility works as normal. We will demonstrate how to do this below.\n",
```

Comment:
"And the `evaluations.run(...)` utility works as normal."

I think we should be explicit about how this should be used. this is cookbook titled "Humanloop RAG Evaluation Walkthrough", and we do state "We will demonstrate how to do this below" right after, yet we don't explicitly show how to use `evaluations.run` with this new `ask_question`

Reply:
as i've said in the other comments, this was incomplete and i chose to move it out because it detracts from the rest of the workbook. It has now been deleted